### DISCUSSION TITLE
Wireguard Kills my Rustdesk Network Connection

### DISCUSSION THREAD

**Discussion by mashme** [rustdesk/rustdesk#6605] [General]
I use my own rustdesk server for long. 
Now I recognized, that the rustdesk client on a windows 10/11 PC does not run with an active wireguard connection. 
When I activate a wireguard vpn tunnel, the rustdesk client closes it network connection. Please see screenshot attached. 
The wireguard and the rustdesk client both work perfect when running alone. The internet traffic beneath the wireguard VPN tunnel ist not blocked by the wireguard configuration. The failure is independent from the rustdesk server I use, It is the same with the public rustdesk network. 
I tested that on three different PCs.  
Can anyone confirm this behavior? 
Is there a configuration workaround to run wireguard and rustdesk at the same time?
Any idea/input is appreciated!
![wireguard kills rustdesk_anonym](https://github.com/rustdesk/rustdesk/assets/8281753/60d09165-4cc3-4579-b25f-7614c2568f0d)

**Comment by subscribernamegoeshere** [comment]
wireguard usually fully routes everything, so its not a split routing, so depending how you use your rustdesk, it disconnects everything except for the target network where you wg into. thats how I know wg at least. split routing etc involves many hackish things still these days with DNS and manually modifying the routes and all.
when the target network you wireguard into, has an internet connection itself outbound and routes your packets your rustdesk should actually recover and come back online again and you could? connect a rustdesk session again thereafter?

**Comment by mashme** [comment]
Thank you @subscribernamegoeshere for your explanation.
Are you sure that wireguard routes all traffic via the vpntunnel? 
There is an wireguard option that looks like you have to activate this "tunnel only" mode: 
![wireguard_tunnel_option](https://github.com/rustdesk/rustdesk/assets/8281753/1e7e7dba-fa97-4ac1-8931-35c2190d7026)
Or is that somethings different? I do not use this option. 
From LANCOM and XCISCO vpn client installations I remember that you can configure split tunneling, if you need it. 
Do you know, is that possible with wireguard too?
If all traffic is routed via th vpn tunnel you are right, the rustdesk connection should find its way through the tunnel with some delay. I will test this, May be I was not patient enough to wait for this.

**Comment by simonmn2006** [comment]
Did you find any solution to this problem , i have extacly the same issue

**Comment by techsupportkoeln** [comment]
it you have the DNS and AllowedIPs line in your config, than all traffic is going through WireGuard.

For a Split Tunnel, you have to delete these lines:

DNS = 192.168.2.1, fritz.box
and to modified these line:
AllowedIPs = 192.168.2.0/24, 128.0.0.0/1

But in this case, only the traffic for the 192.168.2.x Network goes through Wireguard. DNS is not working, or you have to modified your host.etc file.

**Reply by techsupportkoeln** [reply]
and the Option "Blockiere Verkehr außerhalb des Tunnels (Not-Aus)" means, that in case the WireGuard connection goes offline, all traffic will be blocked, so that no traffic can go around the WireGuard Tunnel.
